R/parsing_partitions.R

```diff
@@ -0,0 +1,46 @@
+#path <- getwd()
+#Must have this
+setwd('/home/striker/Desktop/UCLA/BS Research Paper/SideWork')
+
+#install.packages("stringr", repos='http://cran.us.r-project.org')
+library("stringr")
+
+#As a start, try turning the bit that parses the .partitions file into a function 
+#that returns a data frame with three columns, 
+
+#the first being the gene name, 
+#the second being the starting offset, 
+#and the third being the ending offset.
+
+parsing_partitions <- function(file_name) {
+  path <- getwd()
+  
+  fileName <- file_name
+  
+  #using readlines instead of scan and capturing the relevant text from each line using str_match
+  full_name <- paste("/", fileName, sep="")
+  readLinesText <- readLines(paste(path, full_name, sep = ""))
+  print("The output of readLines is:")
+  print(readLinesText)
+  #Now have a vector of the lines
+  
+  #now use str_match
+  #https://regex101.com/r/hyUgZh/1
+  #It is vectorized and so I don't need to write a for loop
+  #HAVE TO USE an extra for every '\'!, and use the website:
+  #https://regex101.com/
+  pattern <- "(\\w+),\\s*(\\w+)\\s*=\\s*(\\d+)\\s*-\\s*(\\d+)"
```

Comment:
List out the expected contents of each capturing group in a comment. (This is generally good practice for any regular expression)